smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/KafkaSource.java

```diff
@@ -86,9 +86,14 @@ public KafkaSource(Vertx vertx,
             kafkaConfiguration.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, config.getKeyDeserializer());
         }
 
+        if (!kafkaConfiguration.containsKey("enable.auto.commit")) {
+            log.disableAutoCommit(config.getChannel());
+            kafkaConfiguration.put("enable.auto.commit", "false");
+        }
+
         String commitStrategy = config
                 .getCommitStrategy()
-                .orElse(Boolean.parseBoolean(kafkaConfiguration.getOrDefault(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "true"))
+                .orElse(Boolean.parseBoolean(kafkaConfiguration.getOrDefault(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false"))
```

Comment:
It looks like **enable.auto.commit** will always have a value now, no need for get**OrDefault** on line 96.

Also , why not use the defined constant ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG instead of the literal **enable.auto.commit** on line 91?

Reply:
Done!